[Buildkite] Add Agent Attribute with Queue Android to Every YML File

ParaskP7 opened the pull request:
### Description
<!-- Take the time to write a good summary. Why is it needed? What does it do? When fixing bugs try to avoid just writing “See original issue” – clarify what the problem was and how you’ve fixed it. -->

This PR adds the `agent` attribute with `queue android` to every YML file missing it, and also, removes the custom meta-data on the scheduled build for dependency analysis ([context](https://github.com/Automattic/buildkite-ci/pull/459#issuecomment-2199589109)).

### Testing information
<!-- This is your opportunity to break out individual scenarios that need testing (when necessary) and/or include a checklist for the reviewer to go through. Consider documenting the following from your own completed testing: devices used, alternate workflows, edge cases, affected areas, critical flows, areas not tested, and any remaining unknowns. Provide feedback on this new section of the PR template through Sept 30, 2024 to Apps Quality; additional context here: p91TBi-b8z-p2#comment-12036 -->

- Make sure CI is green (🟢).

- [x] I have considered if this change warrants release notes and have added them to `RELEASE-NOTES.txt` if necessary. Use the "[Internal]" label for non-user-facing changes.

<!-- Pull request guidelines: https://github.com/woocommerce/woocommerce-android/blob/develop/docs/pull-request-guidelines.md -->

ParaskP7 committed:
CI: Add agent attribute with queue android to every yml file

With that change all the steps in those YML pipelines would build on our
Android queue by default.

FYI: The below '.buildkite/**/*.yml' file were already using this agent:
- pipeline.yml
- schedules/dependency-analysis.yml

For more info see: https://github.com/Automattic/buildkite-ci/pull/459
#issuecomment-2199589109

ParaskP7 committed:
CI: Remove custom meta-data to ci schedule for dependency analysis

This configuration is removed as the goal would be to ultimately use the
'pipeline_file/xyz.yml' metadata key/value instead of this manually set
'scheduled-build/dependency-analysis' meta-data key/value pair.

For more info see: https://github.com/Automattic/buildkite-ci/pull/459
#issuecomment-

dangermattic commented:
<!--
  0 Errors
  1 Warning: PR is not assigned to a milest...
  0 Messages
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">PR is not assigned to a milestone.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

AliSoftware commented on .buildkite/release-pipelines/complete-code-freeze.yml:
```diff
@@ -1,3 +1,6 @@
+agents:
```
Side note while we're at modifying all the `.yml` files: we could consider using the occasion of this PR to add the magic comment that allows IDEs (like VSCode) to recognize those `.yml` files as Buildkite config files (and thus offer syntax highlighting and autocompletion when editing those files in those IDEs):

```suggestion
# yaml-language-server: $schema=https://raw.githubusercontent.com/buildkite/pipeline-schema/main/schema.json
---

agents:
```

Examples of us already using this magic comment in [`tumblr-android`](https://github.com/tumblr/android2/blob/master/.buildkite/pipeline.yml#L1-L2), [`release-toolkit`](https://github.com/wordpress-mobile/release-toolkit/blob/trunk/.buildkite/pipeline.yml#L1-L2), [`buildkite-ci`](https://github.com/Automattic/buildkite-ci/blob/trunk/.buildkite/pipeline.yml#L1-L2)…

Of course if you prefer this can be done in a separate PR, as this is not related to adding `agent` attribute at all (but more about making the edition of those files in VSCode nicer). But just wanted to mention it as you're already editing all the `.yml` files in this PR anyway… 🙂 

PS: Of course this suggestion also applies to https://github.com/wordpress-mobile/WordPress-Android/pull/21023 if you want.

ParaskP7 replied:
Great suggestion @AliSoftware ! ❤️ 

FYI: [Done](https://github.com/woocommerce/woocommerce-android/pull/11863/commits/9c8ac808452e6bd27b624caf310bb8692209af00) + [Done](https://github.com/wordpress-mobile/WordPress-Android/pull/21023/commits/4b2c1b5236f3e285297d5d97e4e795b3f13fe62e)

AliSoftware reviewed (approved):
Looking good :shipit:

ParaskP7 committed:
CI: Add yaml language server schema to allow ides to recognize yml files

FYI: This will offer syntax highlighting and autocompletion when editing
those files in those IDEs (like VSCode).
